subsys/caf/modules/leds.c

```diff
@@ -149,13 +149,13 @@ static void work_handler(struct k_work *work)
 		int32_t next_delay =
 			led->effect->steps[led->effect_step].substep_time;
 
-		k_delayed_work_submit(&led->work, K_MSEC(next_delay));
+		k_work_reschedule(&led->work, K_MSEC(next_delay));
 	}
 }
 
 static void led_update(struct led *led)
 {
-	k_delayed_work_cancel(&led->work);
+	k_work_cancel_delayable(&led->work);
```

Comment:
Looks like another clone-and-own as with https://github.com/nrfconnect/sdk-nrf/pull/4552#discussion_r633154875

Reply:
The CAF leds module is meant to become a common implementation used to remove clone-and-own approach :)
Hopefully it will be used by all of the NCS applications based on Event Manager.

Reply:
Seems like a good plan, maybe it should move out of `subsys/caf/modules` to make that more clear.  Not in this PR, of course.

Reply:
> should move out of `subsys/caf/modules`

Why? This is the common place.

Reply:
Ok; sorry for the noise. I don't use nrf-sdk.  If `caf` means something like "common application framework"  then this makes sense.  (In my experience it means "Complex Ambiguity Function", which would be a pretty specific subsystem.)

Reply:
No worries.
Complex Ambiguity Function sounds scary indeed :)